src/core/pyproxy.ts

```diff
@@ -414,6 +414,14 @@ Module.callPyObjectKwargs = function (
     );
     Py_EXIT();
   } catch (e) {
+    // Emscripten throws "unwind" to stop current code and return to the main event loop.
+    // This is expected behavior and should not be treated as a fatal error.
+    // See: 1) https://github.com/emscripten-core/emscripten/issues/16071
+    //      2) https://github.com/kitao/pyxel/issues/418
+    if (e && e === "unwind") {
+      return;
+    }
```

Comment:
> I think this might be handled better in API.fatal_error (we already handle exit there). Then all of these could change to

Yeah, I also think putting this into `API.fatal_error` would be better. For now, there are some places that assume `API.fatal_error` will not return, so I think we should fix that first.

> Does it actually happen?

[Yes, it happens.](https://github.com/emscripten-core/emscripten/blob/0f494e6c9ef17c1aa52f4b0d2b4efcd38f2f728c/src/library_eventloop.js#L10-L12)

> Does it really not break Python interpreter invariants?

I believe so...? It is just a JS exception.

Reply:
> I believe so...?

Does Pyxel have extra logic to unwind the Python stack?

Reply:
No, pyxel just ignores when [it gets "unwind"](https://github.com/kitao/pyxel/blob/d37290ff07ecc77b06a9af3516f82878113cb3af/wasm/pyxel.js#L421-L423).

Okay, I guess I should do some more experiments and be more careful about this. I'm not sure what can happen on Python interpreter side when Emscripten throws an exception.

Reply:
What I see instead is:
```
Stack (most recent call first):
   File "<exec>", line 8 in g
   File "<exec>", line 5 in f
   File "<exec>", line 10 in <module>
   File "/lib/python311.zip/_pyodide/_base.py", line 310 in run
   File "/lib/python311.zip/_pyodide/_base.py", line 468 in eval_code
```
so there are some "dead frames" that are left on the Python stack.

Reply:
Thanks for the check, then I think we should do

1. unwind Python stack manually, or
2. tell people to not use `simulate_infinite_loop=true` when using `emscripten_set_main_loop`.

@pmp-p Do you have some idea about this? I think pygbag also use [simulate_infinite_loop option](https://github.com/pygame-web/pygbag/blob/a0ea7587ac35ed1920d57ea614464793a024a863/support/__EMSCRIPTEN__.c#L874).

Reply:
Hmm, by the way, can we actually call that "dead frames"? we are kind of conceptually running infinite loops in that frame.

Reply:
> can we actually call that "dead frames"?

I think what you'll see if you call another Python function is that the new frames go on top of the old ones and the old frames are never re entered, and all objects owned by them are leaked. So I believe they are actually leaked frames. But I could be wrong.

Reply:
> I think what you'll see if you call another Python function is that the new frames go on top of the old ones and the old frames are never re entered, and all objects owned by them are leaked.

Yes, indeed the new functions go on top of the old ones.

Okay, but the issue is that I think the old frames are used inside the loop so we can't clear those frames. So maybe we should let users explicitly cancel the loop and maybe block other commands until the loop is running. I think this is going to be more complicated than I thought, so I'll open a separate issue.